### DISCUSSION TITLE
Psalm's generics when using App::make(), app() or resolve()

### DISCUSSION THREAD

**Discussion by daison12006013** [laravel/framework#46907] [General]
Knowing the variable instance really helps me to easily find the methods.

The old way is to force it with `@var`:

```php
/** @var ProfileRepository */
$profile = app(ProfileRepository::class);
```

Although there are cases where we call the method directly, especially when using octane. Constructor args injection isn't good as it is in memory:

```php
// this myMethodHere() is not clickable as the IDE / Editor doesn't know how to locate the defination.
app(ProfileRepository::class)->myMethodHere();
```

---

Meanwhile, when using Psalm plugin in VSCode, it can automatically solve this by just adding the `@template`, I have a modified version of the `app()` docblock and here's how it looks before typing the arrow key:

![image](https://user-images.githubusercontent.com/4581415/235044589-53a366b0-7fc7-4586-b5d4-b3e2e35d40d9.png)

And here's how it looks when hovering over the variable:

![image](https://user-images.githubusercontent.com/4581415/235043163-e9a76968-6ee9-4b1c-97ad-525a5ac104c4.png)

Would it be possible to push my changes through the framework? I don't want to submit the pull request yet.

**Comment by DvDty** [comment]
Interesting why non of the helper files got the type declarations adjustments 🤔

https://github.com/laravel/framework/pull/44545/files
